README.md

```diff
@@ -0,0 +1,82 @@
+# operator-framework-tooling
+
+This repository contains tools for downstreaming OLM software
+
+## OLMv0
+
+The OLMv0 code is downstreamed into the [openshift/operator-framework-olm](https://github.com/openshift/operator-framework-olm) repository from the following upstream repositories:
+
+* [operator-framework/api](https://github.com/operator-framework/api)
+* [operator-framework/operator-lifecycle-manager](https://github.com/operator-framework/operator-lifecycle-manager) - the main repo
+* [operator-framework/operator-registry](https://github.com/operator-framework/operator-registry)
+
+The `operator-lifecycle-manager` repository is the main repository, as it refers to the others. Its `go.mod` file determines what versions of `api` and `operator-registry` are used and downstreamed. Any integration between these three repositories _must_ be done upstream, as that will provide a clean downstreaming experience.
+
+The upstream repositories must also be `go mod tidy` clean.
+
+The downstream repository is a "monorepo" consisting of the above reposities located in a `staging` directory. The upstream commits are cherry-picked with updating comments indicating the source repository and commit SHA.
+
+## OLMv1
+
+The OLMv1 code is downstreamed into the three separate repositories:
+
+* [operator-framework/catalogd](https://github.com/operator-framework/catalogd) -> [openshift/operator-framework-catalogd](https://github.com/openshift/operator-framework-catalogd)
+* [operator-framework/operator-controller](https://github.com/operator-framework/operator-controller) -> [openshift/operator-framework-operator-controller](https://github.com/openshift/operator-framework-operator-controller) - the main repo
+* [operator-framework/rukpak](https://github.com/operator-framework/rukpak) -> [openshift/operator-framework-rukpak](https://github.com/openshift/operator-framework-rukpak)
+
+The upstream commits are downstreamed as a direct mirror (i.e. commit SHAs remain the same) and then merged into the `main` branch.
+
+However `operator-controller` has dependencies on the `catalogd` and `rukpak` repos, and requires that its `go.mod` is updated to reflect the downstream (openshift) repositories.
+1. Use the upstream `go.mod` file in `operator-controller` to determine which commits of `catalogd` and `rukpak` are to be merged.
+2. Synchronize all three downstream repos.
```

Comment:
Is this a well-known procedure?  It it reflecting anything besides k8s bump coordination?

Reply:
Clarify the comment/question? I reviewed the code, and to the best of my knowledge, this is how it works.

Reply:
"synchronize all three downstream repos" means...?

Reply:
I can add some stuff... but this was a non-trivial process!